kernel/arch/riscv/mmap.c

```diff
@@ -5,18 +5,23 @@
 
 #include "paging_int.h"
 
-#define MAX_DMA                                (256 * KB)
-
-STATIC_ASSERT(MAX_DMA <= 16 * MB);
-STATIC_ASSERT((MAX_DMA & (64 * KB - 1)) == 0);
-
 void arch_add_initial_mem_regions()
 {
-   NOT_IMPLEMENTED();
+   /*
+    * Reserve 1MB below kernel's head, where
```

Comment:
Exactly! Since we have a region below Tilck's kernel, why we need a section too?
Also, we have `.data` and `.bss`. Why we need also `.sdata` and `.sbss` ?

Reply:
The`.sdata` section does not have any effect at the moment, because `-Wl,--no-relax` is now used, this option will be removed in the future, which means that the linker will automatically move some commonly accessed global variables to the `.sdata` section. Then use the `gp`(global pointer) register (will always pointing right in the middle of `.sdata`) to access them, which speeds things up.

Reply:
OK, I did some quick research. Correct me if I'm wrong: RISC-V has the concept of a "global pointer" (GP) which can be configured in the CPU and then we use have GP-relative addressing. That reduces the code size compared to a GP-less mode. That's cool!

So, RISC-V doesn't have anything like RIP-relative addressing, right? 
Anyway, got it. The "s*" sections are used for the data accessing using that GP-relative mode. OK, I'm totally cool with that.

Thanks!

Reply:
As I understand it, riscv's gcc option `-mcmodel =medany` is basically equivalent to RIP-relative addressing on x86

Reply:
Aha, that’s interesting; we can play with multiple models.